crates/interpreter/src/exec.rs

```diff
@@ -1011,9 +1012,14 @@ impl<'m> Thread<'m> {
 
     fn push_values(&mut self, values: &[Val]) {
         self.values().extend_from_slice(values);
+        self.label().values_cnt += values.len();
     }
 
     fn pop_value(&mut self) -> Val {
+        // There may be multiple drops in wasm.
```

Comment:
I'm surprised by this logic. Could you provide an example when you hit this issue?